### PR TITLE
feat: Add GPD Upload service infra [PAGOPA-1360]

### PR DESCRIPTION
**Pull request by AngeloCaporaso**
### List of changes

- Blob storage with SFTP enabled
- Lifecycle policy for Blob container
- `STATUS` cosmos container
- GPD APIM v2 with bulk API

### Motivation and context

GPD-Upload requires a blob storage to store the massive data and a cosmos to store the loading status. In addition, GPD-Core implements a new API to load multiple PPs. 
Lifecycle policies allow blobs to be archived and deleted after `N`,`M` days making storage less heavy and expensive.

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/gps-common/03_cosmosdb.tf [hunk 1]
```diff
@@ -84,8 +92,32 @@ locals {
...
+    {
+      name               = "gpd_upload_status",
+      partition_key_path = "/fiscalCode",
+      autoscale_settings = { max_throughput = 1000 }
+    },
+  ]
+}
+
+# cosmosdb container
+module "gpd_cosmosdb_containers" {
```
Comment (AngeloCaporaso) [inline src/domains/gps-common/03_cosmosdb.tf:106]: duplicated

File: src/domains/gps-common/03_gpd_storage_account.tf [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+resource "azurerm_storage_management_policy" "gpd_sa_lifecycle_policy" {
+  storage_account_id = module.gpd_sa.id
+
+  rule {
+    name    = "Gpd upload blob sa policy"
+    enabled = true
+    filters {
+      prefix_match = ["gpd-upload/input", "gpd-upload/output"]
+      blob_types   = ["blockBlob"]
+      # match_blob_index_tag {
```
Comment (AngeloCaporaso) [inline src/domains/gps-common/03_gpd_storage_account.tf:83]: remove if useless

File: src/domains/gps-common/env/weu-prod/terraform.tfvars [hunk 3]
```diff
@@ -134,9 +133,17 @@ reporting_storage_account = {
...
 }
 
-
 geo_replica_enabled                = true
 location_replica                   = "northeurope"
 location_replica_short             = "neu"
 geo_replica_cidr_subnet_postgresql = ["10.2.141.0/24"]
 postgresql_sku_name                = "GP_Gen5_2"
+
+gpd_account_replication_type                                       = "GZRS"
```
Comment (diegolagospagopa) [inline src/domains/gps-common/env/weu-prod/terraform.tfvars:142]: please change to `GRS`

File: src/domains/gps-common/env/weu-uat/terraform.tfvars [hunk 4]
```diff
@@ -103,10 +103,18 @@ cosmos_gpd_payments_db_params = {
...
   }
-
 }
 
 cidr_subnet_gpd_payments_cosmosdb = ["10.1.149.0/24"]
 
 enable_iac_pipeline                   = true
 gpd_payments_sa_delete_retention_days = 0
+
+gpd_account_replication_type                                       = "ZRS"
```
Comment (diegolagospagopa) [inline src/domains/gps-common/env/weu-uat/terraform.tfvars:113]: Please change to GRS

File: src/domains/gps-common/99_variables.tf [hunk 5]
```diff
@@ -355,14 +354,63 @@ variable "reporting_storage_account" {
...
+variable "storage_account_snet_private_link_service_network_policies_enabled" {
+  type        = bool
+  description = "If true, create a private link service"
+  default     = true
+}
+
+variable "cidr_subnet_storage_account" {
+  type        = list(string)
+  description = "Storage account network address space."
+  default     = ["10.1.152.16/29"]
```
Comment (diegolagospagopa) [inline src/domains/gps-common/99_variables.tf:389]: Please don't use default values, but prefer to put into the variables files

File: src/domains/gps-common/03_gpd_storage_account.tf [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  public_network_access_enabled = var.gpd_sa_public_network_access_enabled
+  is_sftp_enabled               = true
+  account_kind                  = "StorageV2"
+  account_tier                  = "Standard"
+  account_replication_type      = var.gpd_account_replication_type
+  access_tier                   = "Hot"
+  is_hns_enabled                = true
+  advanced_threat_protection    = true
+
+  blob_versioning_enabled       = false
```
Comment (diegolagospagopa) [inline src/domains/gps-common/03_gpd_storage_account.tf:17]: Please enable versioning for prod

File: src/domains/gps-common/03_gpd_storage_account.tf [hunk 7]
```diff
@@ -0,0 +1,84 @@
+module "gpd_sa" {
```
Comment (diegolagospagopa) [inline src/domains/gps-common/03_gpd_storage_account.tf:1]: Please can you change the module name to gpd_sa_sftp?

File: src/domains/gps-common/03_gpd_storage_account.tf [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//storage_account?ref=v7.18.0"
+
+  name                = replace("${local.project}-gpd-sa", "-", "")
```
Comment (diegolagospagopa) [inline src/domains/gps-common/03_gpd_storage_account.tf:4]: please change the storage name to 

```suggestion
  name                = replace("${local.project}-gpd-sa-sftp", "-", "")
```

**Review by diegolagospagopa — CHANGES_REQUESTED** [review]

File: src/domains/gps-common/env/weu-prod/terraform.tfvars [hunk 9]
```diff
@@ -134,9 +133,20 @@ reporting_storage_account = {
...
 geo_replica_enabled                = true
 location_replica                   = "northeurope"
 location_replica_short             = "neu"
 geo_replica_cidr_subnet_postgresql = ["10.2.141.0/24"]
 postgresql_sku_name                = "GP_Gen5_2"
+
+# GPD Storage Account SFTP
+gpd_account_replication_type                                       = "GRS"
+cidr_subnet_gpd_storage_account                                    = ["10.1.152.16/29"]
+gpd_sa_blob_versioning_enabled                                     = true
```
Comment (pasqualespica) [inline src/domains/gps-common/env/weu-prod/terraform.tfvars:145]: @AngeloCaporaso should be set to `false` for https://learn.microsoft.com/en-us/azure/storage/blobs/secure-file-transfer-protocol-known-issues

**Review by pasqualespica — CHANGES_REQUESTED** [review]

**Review by pasqualespica — APPROVED** [review]

**Pull request merged**